core/git_mixins/rewrite.py

```diff
@@ -14,6 +17,47 @@ class RewriteMixin():
 
     ChangeTemplate = ChangeTemplate
 
+    def get_commit_chain(self, start, end="HEAD", autosquash=False):
+        commit_chain = [
+            self.ChangeTemplate(orig_hash=entry.long_hash,
+                                do_commit=True,
+                                msg=entry.raw_body,
+                                datetime=entry.datetime,
+                                author="{} <{}>".format(entry.author, entry.email))
+            for entry in self.log(start_end=(start, end), reverse=True)
+            ]
+        if autosquash:
+            i = len(commit_chain) - 1
+            while i > 0:
+                msg = commit_chain[i].msg
+                if msg.startswith("fixup! "):
+                    orig_msg = fixup_command.match(msg).group(1)
+                    j = i - 1
+                    while j >= 0:
+                        if commit_chain[j].msg.startswith(orig_msg):
+                            break
+                        j = j - 1
+                    if j >= 0:
+                        commit_chain.insert(j+1, commit_chain.pop(i))
+                        commit_chain[j].do_commit = False
+                        if i - j >= 2:
+                            # if the fixup commit moves, do not decrease i
+                            continue
+                i = i - 1
```

Comment:
Can you replace `i` and `j` here with variable names that represent the real values you're dealing with?  Looks like it would be something like `commit_idx`.